### DISCUSSION TITLE
v0.8.0

### DISCUSSION THREAD

**Discussion by smaye81** [connectrpc/connect-es#487] [Announcements]
This repository now hosts Connect for Web, but also for Node.js. We changed the name to **Connect-ES** to reflect the universal nature. As part of the reorganization, this release includes new features, but also some **breaking changes**. 

## Connect for Web

This is the first release that shares important bits of code between the web and the Node implementation, and comes with some bug fixes and new features: 

* #334
* #454
* #469
* https://github.com/bufbuild/connect-es/pull/471

If you are using Interceptors, please note the breaking change described in [#471](https://github.com/bufbuild/connect-es/pull/471).

Otherwise, this release is backwards compatible, but we kindly ask you to install `@bufbuild/connect`, and import `ConnectError` and other types from there. In the future, `@bufbuild/connect-web` will only export the Connect and gRPC-web transports.

## Connect for Node.js

The Handlers introduced in the preview are gone! With the new `ConnectRouter`, we can support more frameworks, and this release adds adapters for Express and Fastify:

- https://github.com/bufbuild/connect-es/pull/473
- https://github.com/bufbuild/connect-es/pull/479
- https://github.com/bufbuild/connect-es/pull/474
- https://github.com/bufbuild/connect-es/pull/462

If you have been using Connect for Node, please remove `@bufbuild/connect-core`, install `@bufbuild/connect` instead, and update your imports. See [here](https://github.com/bufbuild/connect-es/blob/199e29e27f8e0d2ab66c3f6c83d50b863cff4de5/packages/connect-node/README.md#connectnodeadapter) how to use ConnectRouter with a Node.js server.

## protoc-gen-connect-web

The code generator plugin has been renamed to `@bufbuild/protoc-gen-connect-es`. The old one is still available, so you don't have to update right away. See [here](https://github.com/bufbuild/connect-es/tree/main/packages/protoc-gen-connect-web) how to update. 

## Other changes
* NodeNext module resolution by @fubhy in #485

<hr /><em>This discussion was created from the release <a href='https://github.com/bufbuild/connect-es/releases/tag/v0.8.0'>v0.8.0</a>.</em>

**Comment by nicole0707** [comment]
I am surprised to discover `Handlers` in @bufbuild/connect-node, Is there any examples to cater for Next.js or any plans to support it? thanks.

**Reply by timostamm** [reply]
Hey Nicole, this release actually removes `Handlers` added in the previous release, and introduces `ConnectRouter` instead. I've updated the description to clarify. We're still updating the docs.

The motivation for that change was to enable support for more frameworks, such as Next.js. Can I ask how you want to use it?

**Reply by nicole0707** [reply]
Thanks @timostamm, we are planning to add gRPC service mocks with Next.js. I checked out the latest version @bufbuild/connect and @bufbuild/connect-express example, it would be better if there is a similar middleware for Next.js. Currently i set up a [custome server](https://nextjs.org/docs/advanced-features/custom-server) with Next.js and @bufbuild/connect-express alternatively.

**Comment by timostamm** [comment]
The docs are now updated with a [step-by-step tutorial](https://connect.build/docs/node/getting-started) and more information for running on Node.js.
